website/docs/r/vapp_firewall_rules.html.markdown

```diff
@@ -0,0 +1,119 @@
+---
+layout: "vcd"
+page_title: "vCloudDirector: vcd_vapp_firewall_rules"
+sidebar_current: "docs-vcd-resource-vapp-firewall-rules"
+description: |-
+  Provides a vCloud Director vApp Firewall resource. This can be used to create, modify, and delete firewall settings and rules.
+---
+
+# vcd\_vapp\_firewall\_rules
+
+Provides a vCloud Director vApp Firewall resource. This can be used to create,
+modify, and delete firewall settings and rules in a [vApp network](/docs/providers/vcd/r/vapp_network.html).
+
+!> **Warning:** Using this resource overrides any existing firewall rules on vApp network.
+
+## Example Usage
+
+```hcl
+resource "vcd_vapp" "web" {
+  name = "web"
+}
+​
+resource "vcd_vapp_network" "vapp-net" {
+  name               = "my-vapp-net"
+  vapp_name          = vcd_vapp.web.name
+  org_network_name   = "my-vdc-int-net"
+  gateway            = "192.168.2.1"
+  netmask            = "255.255.255.0"
+  dns1               = "192.168.2.1"
+​
+  static_ip_pool {
+    start_address = "192.168.2.51"
+    end_address   = "192.168.2.100"
+  }
+}
+
+resource "vcd_vapp_firewall_rules" "vapp_fw" {
+  vapp_id        = vcd_vapp.TestAccVcdVAppForInsert.id
+  network_id     = vcd_vapp_network.vapp-net.id
+  default_action = "drop"
+
+  rule {
+    description      = "drop-ftp-out"
+    policy           = "drop"
+    protocol         = "tcp"
+    destination_port = "21"
+    destination_ip   = "any"
+    source_port      = "any"
+    source_ip        = "10.10.0.0/24"
+  }
+
+  rule {
+    description      = "allow-outbound"
+    policy           = "allow"
+    protocol         = "any"
+    destination_port = "any"
+    destination_ip   = "any"
+    source_port      = "any"
+    source_ip        = "10.10.0.0/24"
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `org` - (Optional; *v2.0+*) The name of organization to use, optional if defined at provider level. Useful when connected as sysadmin working across different organisations.
+* `vdc` - (Optional; *v2.0+*) The name of VDC to use, optional if defined at provider level.
+* `vapp_id` - (Required) The identifier of vApp.
```

Comment:
vApp - please add a link to doc page here too.

Reply:
added

Reply:
added